InterviewPrep/chapter01/question0103.cs

```diff
@@ -1,19 +1,98 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Question0103.cs" company="">
+//   
+// </copyright>
+// <summary>
+//   The question 0103.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
 
 /**
     Given two strings, write a method to decide if one is a permutation of the other.
 **/
-namespace InterviewPrep.chapter01
+namespace InterviewPrep.Chapter01
 {
-    class question0103
+    using System;
+
+    /// <summary>
+    ///     The question 0103.
+    /// </summary>
+    public class Question0103
     {
-        static bool IsPermutation(String s1, String s2)
+        /// <summary>
+        /// The is permutation. 
+        /// </summary>
+        /// <param name="s1">
+        /// The s 1. 
+        /// </param>
+        /// <param name="s2">
+        /// The s 2. 
+        /// </param>
+        /// <returns>
+        /// The <see cref="bool"/>. 
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="array"/> is null.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        /// One or more elements in <paramref name="array"/> do not implement the
+        ///     <see cref="T:System.IComparable`1"/> generic interface.
+        /// </exception>
+        public static bool IsPermutation(string s1, string s2)
         {
+            if (CornerCase(s1, s2))
```

Comment:
Not sure why when s.length==1 will this lead to a true without inspecting equality? One thing i suggest is to try to dry run a few test cases to check these edge cases.